scripts/linters/pylint_extensions.py

```diff
@@ -2139,6 +2140,71 @@ def visit_classdef(self, node):
             self.add_message('no-dunder-metaclass', node=node)
 
 
+class DisallowHandlerWithoutSchema(checkers.BaseChecker):
+    """Custom pylint checker prohibiting handlers which do not have schema
+    defined within the class.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    name = 'disallow-handlers-without-schema'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'Please add schema in URL_ARGS_PATH_SCHEMA for %s class. \nVisit '
+            'https://github.com/oppia/oppia/wiki/Validation-of-handler-args'
+            'to learn how to write schema for handlers.',
+            'no-schema-for-url-path-elements',
+            'Enforce writing schema for url path arguments of handler class.'
+        ),
+        'C0036': (
+            'Please add schema in HANDLER_ARGS_SCHEMA for %s class. \nVisit '
+            'https://github.com/oppia/oppia/wiki/Validation-of-handler-args'
+            'to learn how to write schema for handlers.',
+            'no-schema-for-handler-args',
+            'Enforce writing schema for request arguments of handler class.'
+        )
+    }
+
+    def check_parent_class_is_basehandler(self, node):
+        """Checks whether a class is child class of BaseHandler.
+
+        Returns:
+            bool. Whether a class is child class of BaseHandler.
+        """
+        # For checking one level of inheritance.
+        parent_class_is_basehandler = 'base.BaseHandler' in node.basenames
```

Comment:
Why do we need to separately check in node.basenames if we are checking for all the ancestors below? I don't think `node.ancestors` would be a costly operation which you want to avoid by doing this first. This is not needed right?

Reply:
Removed, ptal